sources/web/datalab/polymer/components/table-details/table-details.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * Table details pane element for Datalab.
+ * This element is designed to be displayed in a side bar that displays more
+ * information about a selected BigQuery table.
+ */
+class TableDetailsElement extends Polymer.Element {
+
+  /**
+   * Id for table whose details to show.
+   */
+  public tableId: string;
+
+  private _table: gapi.client.bigquery.BigqueryTable | null;
+  private _busy = false;
+  private readonly DEFAULT_MODE = 'NULLABLE';
+
+  static get is() { return 'table-details'; }
+
+  static get properties() {
+    return {
+      _table: {
+        type: Object,
+        value: null,
+      },
+      creationTime: {
+        computed: '_computeCreationTime(_table)',
+        type: String,
+      },
+      lastModifiedTime: {
+        computed: '_computeLastModifiedTime(_table)',
+        type: String,
+      },
+      longTermTableSize: {
+        computed: '_computeLongTermTableSize(_table)',
+        type: String,
+      },
+      numRows: {
+        computed: '_computeNumRows(_table)',
+        type: String,
+      },
+      tableId: {
+        observer: '_tableIdChanged',
+        type: String,
+        value: '',
+      },
+      tableSize: {
+        computed: '_computeTableSize(_table)',
+        type: String,
+      },
+    };
+  }
+
+  _tableIdChanged() {
+    const matches = this.tableId.match(/^(.*):(.*)\.(.*)$/);
+    if (matches && matches.length === 4) { // The whole string is matched as first result
+      this._busy = true;
+      const projectId = matches[1];
+      const datasetId = matches[2];
+      const tableId = matches[3];
+
+      GapiManager.getTableDetails(projectId, datasetId, tableId)
+        .then((result) => {
+          this._table = JSON.parse(result.body) as gapi.client.bigquery.BigqueryTable;
```

Comment:
Instead of result here you can call it response and declare its type as HttpResponse<gapi.client.bigquery.BigqueryTable> (or GetTableResponse assuming you change the response name to that as suggested elsewhere), then you can directly access response.result as the table details rather than calling JSON.parse.